pravega/pravega-keycloak
# Issue 12: Modify build for pushing publications to Maven Central

**Change log description**  
Modify build for pushing publications to Maven Central. 

**Purpose of the change**  
Resolves #12 

**What the code does**  

* Modify build process to push to Maven Central. 
* Extracts out release signing and publishing to a separate Gradle file `maven-publish.gradle`. 

**How to Test**

1. Publish to: 
  * LocalRepo: `./gradlew publish -PpublishRepo=localFileRepo`
  * Artifactory: `./gradlew publish -PpublishRepo=jcenter -PpublishUsername=shardr -PpublishUrl=http://<your-artifactory-instance>:8081/artifactory/<project>`
2. Verify the published artifacts

To sign the builds, use the signing options from the last command below in step 1. 

Commands for publishing the release:
  * JCenter Snapshots: `./gradlew clean publish -PpublishRepo=jcenterSnapshot -PpublishUsername=<username> -PpublishPassword=<password>
  * Maven Central Snapshots: `./gradlew clean publish -PdoSigning=true -Psigning.password=<signing-password> -PpublishRepo=mavenCentral -PpublishUsername=rsharda -PpublishPassword=<sonatype-password>`
